crates/pool-manager/src/rollup.rs

```diff
@@ -0,0 +1,257 @@
+        // This is typically simpler and more direct
+        self.order_indexer
+            .get_all_orders_with_parked()
+            .into_all_orders()
+    }
+}
+
+// Implement the PoolManagerCommon trait methods
+impl<V, GS> PoolManagerCommon for RollupPoolManager<V, GS>
+where
+    V: OrderValidatorHandle<Order = AllOrders> + Unpin,
+    GS: BlockSyncConsumer
+{
+    type GlobalSync = GS;
+    type Validator = V;
+
+    // Use macro to avoid duplication of getter methods
+    impl_common_getters!(RollupPoolManager<V, GS>, V, GS);
+
+    fn on_command(&mut self, cmd: OrderCommand) {
+        use angstrom_types::orders::OrderOrigin;
+        use telemetry_recorder::telemetry_event;
+
+        match cmd {
+            OrderCommand::NewOrder(origin, order, validation_response) => {
+                let blocknum = self.global_sync.current_block_number();
+                telemetry_event!(blocknum, origin, order.clone());
+
+                self.order_indexer
+                    .new_rpc_order(OrderOrigin::External, order, validation_response)
+            }
+            OrderCommand::CancelOrder(req, receiver) => {
+                let blocknum = self.global_sync.current_block_number();
+                telemetry_event!(blocknum, req.clone());
+
+                let res = self.order_indexer.cancel_order(&req);
+                // In rollup mode, no need to broadcast cancellation to peers
+                let _ = receiver.send(res);
+            }
+            OrderCommand::PendingOrders(from, receiver) => {
+                let res = self.order_indexer.pending_orders_for_address(from);
+                let _ = receiver.send(res.into_iter().map(|o| o.order).collect());
+            }
+            OrderCommand::OrderStatus(order_hash, tx) => {
+                let res = self.order_indexer.order_status(order_hash);
+                let _ = tx.send(res);
+            }
+            OrderCommand::OrdersByPool(pool_id, location, tx) => {
+                let res = self.order_indexer.orders_by_pool(pool_id, location);
+                let _ = tx.send(res);
+            }
+        }
+    }
+
+    fn on_pool_events(&mut self, orders: Vec<PoolInnerEvent>, waker: impl Fn() -> Waker) {
+        for order in orders {
+            match order {
+                PoolInnerEvent::Propagation(_order) => {
+                    // In rollup mode, no need to broadcast orders to peers
+                    // since there's no networking
+                }
+                PoolInnerEvent::BadOrderMessages(_o) => {
+                    // In rollup mode, we don't have networking, so no
+                    // reputation changes
+                }
+                PoolInnerEvent::HasTransitionedToNewBlock(block) => {
+                    self.global_sync
+                        .sign_off_on_block(MODULE_NAME, block, Some(waker()));
+                }
+                PoolInnerEvent::None => {}
+            }
+        }
+    }
+}
```

Comment:
could simplify this by removing `match` and only handle `PoolInnerEvent::HasTransitionedToNewBlock`, but I think the no-ops also serve as documentation